src/Test/Html/Events.elm

```diff
@@ -0,0 +1,149 @@
+module Test.Html.Events
+    exposing
+        ( Event(..)
+        , simulate
+        )
+
+{-|
+
+This module allows you to simulate events on Html nodes, the Msg generated
+by the event is returned so you can test it
+
+@docs Event, simulate
+
+-}
+
+import ElmHtml.InternalTypes exposing (ElmHtml(NodeEntry))
+import Json.Decode exposing (decodeString)
+import Json.Encode exposing (bool, encode, object, string)
+import Native.HtmlAsJson
+import Test.Html.Query as Query
+import Test.Html.Query.Internal as QueryInternal
+
+
+{-| Event constructors to simulate events
+-}
+type Event
```

Comment:
So, we cannot create a `Dict Event Json.Value` because `Event` is not comparable. But I don't see a problem in left the string name of the event there, and create the custom `Event` type here, since for testing it is important, but for parsing maybe not.

I've opened a PR: https://github.com/eeue56/elm-html-in-elm/pull/6